service/src/main/kotlin/io/provenance/onboarding/domain/usecase/cee/ExecuteContract.kt

```diff
@@ -0,0 +1,113 @@
+package io.provenance.onboarding.domain.usecase.cee
+
+import com.google.protobuf.Message
+import io.provenance.core.KeyType
+import io.provenance.onboarding.domain.cee.ContractParser
+import io.provenance.onboarding.domain.cee.ContractService
+import io.provenance.onboarding.domain.provenance.Provenance
+import io.provenance.onboarding.domain.usecase.AbstractUseCase
+import io.provenance.onboarding.domain.usecase.cee.model.ExecuteContractRequest
+import io.provenance.onboarding.domain.usecase.common.model.TxResponse
+import io.provenance.onboarding.domain.usecase.common.originator.GetOriginator
+import io.provenance.onboarding.domain.usecase.provenance.account.GetAccount
+import io.provenance.onboarding.frameworks.provenance.utility.ProvenanceUtils
+import io.provenance.scope.contract.annotations.Input
+import io.provenance.scope.encryption.model.DirectKeyRef
+import io.provenance.scope.encryption.util.toJavaPrivateKey
+import io.provenance.scope.encryption.util.toJavaPublicKey
+import io.provenance.scope.sdk.Affiliate
+import io.provenance.scope.sdk.Client
+import io.provenance.scope.sdk.ClientConfig
+import io.provenance.scope.sdk.SharedClient
+import java.net.URI
+import io.provenance.scope.contract.spec.P8eContract
+import java.security.KeyPair
+import java.util.concurrent.TimeUnit
+import kotlin.reflect.full.functions
+import mu.KotlinLogging
+import org.springframework.stereotype.Component
+
+private val log = KotlinLogging.logger { }
+
+@Component
+class ExecuteContract(
+    private val getOriginator: GetOriginator,
+    private val contractService: ContractService,
+    private val provenanceService: Provenance,
+    private val getAccount: GetAccount,
+    private val contractParser: ContractParser,
+) : AbstractUseCase<ExecuteContractRequest, TxResponse>() {
+
+    override suspend fun execute(args: ExecuteContractRequest): TxResponse {
+        val utils = ProvenanceUtils()
+        val account = getAccount.execute(args.config.account)
+        val originator = getOriginator.execute(args.config.account.originatorUuid)
+        val affiliate = Affiliate(
+            signingKeyRef = DirectKeyRef(KeyPair(originator.keys[KeyType.SIGNING_PUBLIC_KEY].toString().toJavaPublicKey(), originator.keys[KeyType.SIGNING_PRIVATE_KEY].toString().toJavaPrivateKey())),
+            encryptionKeyRef = DirectKeyRef(KeyPair(originator.keys[KeyType.SIGNING_PUBLIC_KEY].toString().toJavaPublicKey(), originator.keys[KeyType.SIGNING_PRIVATE_KEY].toString().toJavaPrivateKey())),
+            args.config.account.partyType,
+        )
+
+        val sharedClient = SharedClient(
+            ClientConfig(
+                mainNet = !args.config.account.isTestNet,
+                cacheJarSizeInBytes = 4L * 1024 * 1024, // ~ 4 MB,
```

Comment:
curious, what is this 4 MB limit? what happens if Jar size is larger than 4 MB?

Reply:
great question - not honestly sure. This was the default that `service-p8e-los` used, many others use `0L` (assuming 0 means no limit?)... the documentation on this is lacking.